boms/cloud-oss-bom/pom.xml

```diff
@@ -69,207 +72,202 @@
-        <artifactId>grpc-testing</artifactId>
-        <version>${io.grpc.version}</version>
+        <groupId>com.google.api-client</groupId>
+        <artifactId>google-api-client-xml</artifactId>
+        <version>${google.api-client.version}</version><!-- {x-version-update:google-api-client:current} -->
       </dependency>
 
-      <!-- google-cloud-java from https://github.com/googleapis/java-cloud-bom-->
-      <dependency>
-        <groupId>com.google.cloud</groupId>
-        <artifactId>google-cloud-bom</artifactId>
-        <version>${google.cloud.java.version}</version>
-        <type>pom</type>
-        <scope>import</scope>
-      </dependency>
+
 
       <dependency>
-        <groupId>com.google.api</groupId>
-        <artifactId>api-common</artifactId>
-        <version>1.10.4</version>
+        <groupId>com.google.api.grpc</groupId>
+        <artifactId>proto-google-iam-v1</artifactId>
+        <version>${iam.version}</version>
       </dependency>
+
+      <!-- Following test deps are kept to keep them consistent with versions above -->
       <dependency>
-        <groupId>com.google.api</groupId>
-        <artifactId>gax</artifactId>
-        <version>${gax.version}</version>
+        <groupId>com.google.api.grpc</groupId>
+        <artifactId>grpc-google-iam-v1</artifactId>
+        <version>${iam.version}</version>
       </dependency>
       <dependency>
         <groupId>com.google.api</groupId>
         <artifactId>gax-grpc</artifactId>
         <version>${gax.version}</version>
-      </dependency>
-      <dependency>
-        <groupId>com.google.api</groupId>
-        <artifactId>gax-httpjson</artifactId>
-        <version>${gax.httpjson.version}</version>
-      </dependency>
-      <dependency>
-        <groupId>com.google.auth</groupId>
-        <artifactId>google-auth-library-bom</artifactId>
-        <version>${auth.version}</version>
-        <type>pom</type>
-        <scope>import</scope>
+        <classifier>testlib</classifier>
       </dependency>
       <dependency>
         <groupId>com.google.cloud</groupId>
-        <artifactId>google-cloud-core-bom</artifactId>
-        <version>${google.cloud.core.version}</version>
-        <type>pom</type>
-        <scope>import</scope>
-      </dependency>
-      <dependency>
-        <groupId>com.google.api.grpc</groupId>
-        <artifactId>proto-google-common-protos</artifactId>
-        <version>${common.protos.version}</version>
-      </dependency>
-      <dependency>
-        <groupId>com.google.api.grpc</groupId>
-        <artifactId>grpc-google-common-protos</artifactId>
-        <version>${common.protos.version}</version>
-      </dependency>
-      <dependency>
-        <groupId>com.google.api.grpc</groupId>
-        <artifactId>proto-google-iam-v1</artifactId>
-        <version>${iam.protos.version}</version>
+        <artifactId>google-cloud-core</artifactId>
+        <version>${google.core.version}</version>
+        <type>test-jar</type>
```

Comment:
Linkage Checker is not handling the google-cloud-core artifact that appears multiple times with different classifiers.

The error is shown as if google-cloud-core is missing.

```
Error:  testForNewLinkageErrors(com.google.cloud.MaximumLinkageErrorsTest)  Time elapsed: 161.994 s  <<< FAILURE!
java.lang.AssertionError: 
Baseline BOM: com.google.cloud:libraries-bom:20.9.0
Newly introduced problems:
Class com.google.cloud.ServiceOptions is not found;
  referenced by 24 class files
    com.google.cloud.ServiceOptionsTest (com.google.cloud:google-cloud-core:1.95.4)
    com.google.cloud.BaseWriteChannelTest (com.google.cloud:google-cloud-core:1.95.4)
    com.google.cloud.SerializationTest (com.google.cloud:google-cloud-core:1.95.4)
    com.google.cloud.bigquery.BigQueryOptions (com.google.cloud:google-cloud-bigquery:1.137.1)
```

Reply:
The dependency management includes classifier.

pom.xml:

```
suztomo@suztomo-macbookpro44 maven-dependency-management-classifier % cat pom.xml        
<?xml version="1.0" encoding="UTF-8"?>

<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>org.example</groupId>
  <artifactId>maven-dependency-management-classifier</artifactId>
  <version>1.0-SNAPSHOT</version>

  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <maven.compiler.source>1.8</maven.compiler.source>
    <maven.compiler.target>1.8</maven.compiler.target>
  </properties>

  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>com.google.cloud</groupId>
        <artifactId>google-cloud-core</artifactId>
        <version>1.96.0</version><!-- {x-version-update:google-cloud-core:current} -->
      </dependency>
    </dependencies>
  </dependencyManagement>

  <dependencies>
    <dependency>
      <groupId>com.google.cloud</groupId>
      <artifactId>google-cloud-core</artifactId>
      <type>test-jar</type>
    </dependency>
  </dependencies>
</project>
```

The build fails:

```
suztomo@suztomo-macbookpro44 maven-dependency-management-classifier % mvn dependency:tree
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.version' for com.google.cloud:google-cloud-core:test-jar is missing. @ line 28, column 17
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project org.example:maven-dependency-management-classifier:1.0-SNAPSHOT (/Users/suztomo/maven-dependency-management-classifier/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for com.google.cloud:google-cloud-core:test-jar is missing. @ line 28, column 17
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
```

Reply:
Interestingly, the two Maven coordinates refer to the same JAR file.

<img width="1434" alt="Screen Shot 2021-07-30 at 16 07 44" src="https://user-images.githubusercontent.com/28604/127706555-ef6b0407-53ff-4e6e-bf09-516d38b10c92.png">

<img width="1425" alt="Screen Shot 2021-07-30 at 16 08 35" src="https://user-images.githubusercontent.com/28604/127706556-750a4c2e-850b-4056-8907-3534f36de0f8.png">